ground/uavobjgenerator/uavobjectparser.cpp

```diff
@@ -169,6 +177,24 @@ void UAVObjectParser::calculateAllIds()
     foreach (ObjectInfo *item, objInfo) {
         calculateID(item);
     }
+
+    /* Sort the object info list by ID, now that they're all defined */
+    std::sort(objInfo.begin(), objInfo.end(), [](ObjectInfo *o1, ObjectInfo *o2) {
+            return o2->id < o1->id;
+            });
+
+    /* This is a 64 bit hash to have a bit more protection against collisions

```

Comment:
It's not yet.  The only purpose right now is to cause a file which is marked as a dependency to change if the actual contents have changed.  But the intention is to perhaps replace the existing sha1 hash with a hash generated here (perhaps a SHA-1, perhaps something more like this), so that A) we're no longer dependent on formatting/versions when calculating the uav hash, and B) we know the hash earlier in the build process and some of the weirder glue in the makefile to handle uavo hashes can be removed.